kroxylicious/src/test/java/io/kroxylicious/proxy/internal/KafkaProxyFrontendHandlerTest.java

```diff
@@ -0,0 +1,224 @@
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+
+class KafkaProxyFrontendHandlerTest {
+
+    public static final String SNI_HOSTNAME = "external.example.com";
+    public static final String CLUSTER_HOST = "internal.example.org";
+    public static final int CLUSTER_PORT = 9092;
+    EmbeddedChannel inboundChannel;
+    EmbeddedChannel outboundChannel;
+
+    int corrId = 0;
+
+    private void writeRequest(short apiVersion, ApiMessage body) {
+        var apiKey = ApiKeys.forId(body.apiKey());
+
+        int downstreamCorrelationId = corrId++;
+
+        short headerVersion = apiKey.requestHeaderVersion(apiVersion);
+        RequestHeaderData header = new RequestHeaderData()
+                .setRequestApiKey(apiKey.id)
+                .setRequestApiVersion(apiVersion)
+                .setClientId("client-id")
+                .setCorrelationId(downstreamCorrelationId);
+
+        inboundChannel.writeInbound(new DecodedRequestFrame<>(apiVersion, corrId, true, header, body));
+    }
+
+    @BeforeEach
+    public void buildChannel() {
+        inboundChannel = new EmbeddedChannel();
+        corrId = 0;
+    }
+
+    @AfterEach
+    public void closeChannel() {
+        inboundChannel.close();
+    }
+
+    public static List<Arguments> provideArgsForExpectedFlow() {
+        var result = new ArrayList<Arguments>();
+        boolean[] tf = { true, false };
+        for (boolean sslConfigured : tf) {
+            for (boolean haProxyConfigured : tf) {
+                for (boolean saslOffloadConfigured : tf) {
+                    for (boolean sendApiVersions : tf) {
+                        for (boolean sendSasl : tf) {
+                            result.add(Arguments.of(sslConfigured, haProxyConfigured, saslOffloadConfigured, sendApiVersions, sendSasl));
+                        }
+                    }
+                }
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Test the normal flow, in a number of configurations.
+     * @param sslConfigured Whether SSL is configured
+     * @param haProxyConfigured
+     * @param saslOffloadConfigured
+     * @param sendApiVersions
+     * @param sendSasl
+     */
+    @ParameterizedTest
+    @MethodSource("provideArgsForExpectedFlow")
+    public void expectedFlow(boolean sslConfigured,
+                             boolean haProxyConfigured,
+                             boolean saslOffloadConfigured,
+                             boolean sendApiVersions,
+                             boolean sendSasl) {
+
+        var dp = new SaslDecodePredicate(saslOffloadConfigured);
+        ArgumentCaptor<NetFilter.NetFilterContext> valueCapture = ArgumentCaptor.forClass(NetFilter.NetFilterContext.class);
+        var filter = mock(NetFilter.class);
+        doAnswer(i -> {
+            NetFilter.NetFilterContext ctx = i.getArgument(0);
+            assertEquals(sslConfigured ? SNI_HOSTNAME : null, ctx.sniHostname());
```

Comment:
```suggestion
           if (sslConfigured) {
                       assertEquals(sslConfigured, ctx.sniHostname());
           } else {
                       assertNull(ctx.sniHotname());
           }
```

I think that's easier to read and more explicit about what the expectations are.